src/QueryEditor/QueryEditor.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { css } from '@emotion/css';
+import { QueryEditorProps } from '@grafana/data';
+import { CentreonDataSource } from '../centreonDataSource';
+import { CentreonMetricOptions, defaultQuery, MyQuery } from '../@types/types';
+import { debounce, defaults } from 'lodash';
+import { QueryEditorModeSwitcher } from './QueryEditorModeSwitcher';
+import { EMode } from './EMode';
+import { VisualCentreonEditor } from './VisualCentreonEditor';
+import { RawCentreonQueryEditor } from './RawCentreonQueryEditor';
+
+type Props = QueryEditorProps<CentreonDataSource, MyQuery, CentreonMetricOptions>;
+
+export const QueryEditor: React.FC<Props> = (props: Props) => {
+  const { query, onRunQuery, onChange: pOnchange, datasource } = props;
+  const state = defaults(query, defaultQuery);
+
+  const [loading, setLoading] = useState(false);
+  const onRunQueryDebounced = useCallback(() => debounce(onRunQuery, 300)(), [onRunQuery]);
+
+  const onChange = useCallback((value: MyQuery) => {
+    pOnchange(value);
+    // onChange from parent seems to change everytime, and so create an infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    onRunQueryDebounced();
+  }, [query, onRunQueryDebounced]);
+
+  const mode = state.mode ?? EMode.VISUAL;
+  const CurrentEditor =
```

Comment:
variables are named using camelCase (even if it contains JSX). PascalCase is reserved for Components